server/src/infra/repositories/metadata.repository.ts

```diff
@@ -47,6 +48,17 @@ export class MetadataRepository implements IMetadataRepository {
 
     this.logger.log('Importing geodata to database from file');
 
+    this.databaseRepository.withLock(DatabaseLock.GeodataImport, this.importGeodata);
```

Comment:
This should be awaited, the callback should be () => this.importGeodata()

Reply:
Oops, forgot to await that. I think passing the method works fine though, any particular reason to wrap it like that?

Reply:
Otherwise you can't use "this" in the method correctly. It probably isn't working correctly right now.

Reply:
Oh... yeah, that seems like a good reason to change it.